ament_cppcheck/ament_cppcheck/main.py

```diff
@@ -116,6 +116,13 @@ def main(argv=sys.argv[1:]):
                 'set the AMENT_CPPCHECK_ALLOW_1_88 environment variable to override this.',
                 file=sys.stderr,
             )
+
+            if args.xunit_file:
+                report = {input_file: [] for input_file in files}
+                write_xunit_file(args.xunit_file, report, time.time() - start_time,
+                                 skip='cppcheck 1.88 performance issues')
+                return 0
```

Comment:
It is pretty obscure that the return code changes based on the presence of the `--xunit-file` argument but I see why it is done :disappointed:

Reply:
I don't disagree with you. My thought process is that that if the xunit file was specified, that is meant to be the medium through which results are expressed, and we need to return `0` if we want those results to be presented on the buildfarm.

We could make the return code always zero, but that seems misleading. This seems like an okay compromise.